backend/db/mongo.go

```diff
@@ -57,17 +59,28 @@ func (c *Database) FindAllArticles() ([]Article, error) {
 	}
 	var articles []Article
 
+	numObtained := 0
 	for cursor.Next(context.TODO()) {
```

Comment:
I would prefer if we have a loop where we put numObtained in the condition, saves us from the extra if statement. And assuming we have this in a separate function to get rid of the -1 special handling all will be good
```go
for cursor.Next(context.TODO()) && numObtained < numArticles {
.
.
}

```